### DISCUSSION TITLE
Get block under player

### DISCUSSION THREAD

**Discussion by Codestech1** [Minestom/Minestom#1674] [Help] [answered]
Hi, how can i get block under player in best way?

**Comment by iam4722202468** [comment, marked as answer]
There isn't one easy way to do this. There can be a situation where a player has no block directly under them based on their position, but is still standing on a block because the corner of their bounding box is on a block. Since the player's x and z bounding box sizes are under 1, you should be able to take the bottom corners of the bounding box, add the player's position, then subtract a small value like 0.1 from the y value and get the 4 blocks under the player.
